addons/point_of_sale/static/src/app/store/pos_store.js

```diff
@@ -1637,7 +1637,7 @@ export class PosStore extends Reactive {
     getReceiptHeaderData(order) {
         return {
             company: this.company,
-            cashier: _t("Served by %s", this.get_cashier()?.name),
+            cashier: _t("Served by %s", order.employee_id?.name || this.get_cashier()?.name),
```

Comment:
employee_id is not in point_of_sale. create a method `getCashierName` in point_of_sale then override it in pos_hr.

Reply:
@caburj I noticed this. But as it's used in export_for_printing in point_of_sale, I thought it could be correct.

Reply:
I'm sticking with my request. You may also include in this PR the change to move the use of `employee_id` in `export_for_printing` to `pos_hr`.

Reply:
@caburj I modified the code.